lib/spack/spack/user_environment.py

```diff
@@ -59,33 +62,46 @@ def unconditional_environment_modifications(view):
     return env
 
 
-def environment_modifications_for_spec(spec, view=None):
+@contextlib.contextmanager
+def nullcontext(repo):
+    yield
+
+
+def environment_modifications_for_spec(in_spec, view=None):
     """List of environment (shell) modifications to be processed for spec.
 
     This list is specific to the location of the spec or its projection in
     the view."""
-    spec = spec.copy()
-    if view and not spec.external:
-        spec.prefix = prefix.Prefix(view.view().get_projection_for_spec(spec))
-
-    # generic environment modifications determined by inspecting the spec
-    # prefix
-    env = environment.inspect_path(
-        spec.prefix,
-        prefix_inspections(spec.platform),
-        exclude=environment.is_system_path
-    )
-
-    # Let the extendee/dependency modify their extensions/dependents
-    # before asking for package-specific modifications
-    env.extend(
-        build_env.modifications_from_dependencies(
-            spec, context='run'
+    spec = in_spec.copy()
+    pkgrepopath = os.path.join(
+        spack.store.layout.build_packages_path(in_spec),
+        spec.namespace)
+    context = nullcontext
+    if not spec.external:
+        context = spack.repo.swap_in_additional_repository
+    with context(pkgrepopath):
```

Comment:
This all looks good, but it's hard to grok. Can you add block comments to it please?

Changing to
```
if spec.external:
    context = nullcontext
else:
    context = spack.repo.swap_in_additional_repository
```
might also make it clearer. (Note, in python you can define a variable inside an if-block and then use it outside the block. Weird, but sometimes useful.)